src/Platform/Microsoft.Testing.Extensions.AzureDevOps/AzureDevOpsReporter.cs

```diff
@@ -119,10 +119,8 @@ private async Task WriteExceptionAsync(string? explanation, Exception? exception
         }
 
         string stackTrace = exception.StackTrace;
-        int index = stackTrace.IndexOfAny(NewlineCharacters);
-        string firstLine = index == -1 ? stackTrace : stackTrace.Substring(0, index);
-
-        (string Code, string File, int LineNumber)? location = GetStackFrameLocation(firstLine);
+        string[] lines = stackTrace.Split(NewlineCharacters, StringSplitOptions.RemoveEmptyEntries);
+        (string Code, string File, int LineNumber)? location = lines.Select(GetStackFrameLocation).FirstOrDefault(location => location is not null);
```

Comment:
This will (in GetStackFrameLocation) find all files that exist (and will check for every line in stack trace), and then it will take just the first one found. 

2 issues with that:
1) we do more checks of the filesystem than necessary
2) won't work with deterministic build because the paths won't ever exist.

but both are okay, I will merge this PR for the other changes, and fix it in #5481 where I already handle deterministic build.